src/main/java/org/mskcc/cbio/oncokb/web/rest/CreateAssessment.java

```diff
@@ -0,0 +1,84 @@
+package org.mskcc.cbio.oncokb.web.rest;
+
+import com.amazonaws.services.appstream.model.Application;
+import com.google.cloud.recaptchaenterprise.v1.RecaptchaEnterpriseServiceClient;
+import com.google.recaptchaenterprise.v1.Assessment;
+import com.google.recaptchaenterprise.v1.CreateAssessmentRequest;
+import com.google.recaptchaenterprise.v1.Event;
+import com.google.recaptchaenterprise.v1.ProjectName;
+import java.io.IOException;
+
+import org.mskcc.cbio.oncokb.config.application.ApplicationProperties;
+import org.mskcc.cbio.oncokb.config.application.RecaptchaProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.xml.bind.ValidationException;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.RequestParam;
+
+@Component
+public class CreateAssessment {
+  
+  private static final Logger LOGGER = LoggerFactory.getLogger(CreateAssessment.class);
+
+  static RecaptchaProperties recaptchaProperties;
+
+  public CreateAssessment(ApplicationProperties applicationProperties){
+      this.recaptchaProperties = applicationProperties.getRecaptcha();
+  }
+
+  /**
+   * Create an assessment to analyze the risk of an UI action. Assessment approach
+   * is the same for
+   * both 'score' and 'checkbox' type recaptcha site keys.
+   * @return
+   * @throws ValidationException
+   */
+  public static ResponseEntity<String> createAssessment(HttpServletRequest request)
+      throws IOException, ValidationException {
+
+    String recaptchaToken = request.getHeader("g-recaptcha-response");
+    if (recaptchaToken == null) {
+      throw new ValidationException("Unable to retrieve recaptcha token. Please try again.");
+    }
+
+    try (RecaptchaEnterpriseServiceClient client = RecaptchaEnterpriseServiceClient.create()) {
+
+      // Set the properties of the event to be tracked.
+      Event event = Event.newBuilder().setSiteKey(recaptchaProperties.getRecaptchaSiteKey()).setToken(recaptchaToken).build();
+
+      // Build the assessment request.
+      CreateAssessmentRequest createAssessmentRequest = CreateAssessmentRequest.newBuilder()
+          .setParent(ProjectName.of(recaptchaProperties.getRecaptchaProjectId()).toString())
+          .setAssessment(Assessment.newBuilder().setEvent(event).build())
+          .build();
+
+      Assessment response = client.createAssessment(createAssessmentRequest);
+
+      client.close();
+
+      // Check if the token is valid.
+      if (response.getTokenProperties().getValid() && response.getRiskAnalysis().getScore() >= Long.valueOf(recaptchaProperties.getRecaptchaThreshold())) {
```

Comment:
```suggestion
      if (response.getTokenProperties().getValid() && response.getRiskAnalysis().getScore() >= Float.valueOf(recaptchaProperties.getRecaptchaThreshold())) {
```
We should use `float` type here since Long is for whole numbers. The threshold can only be from 0.0 to 1.0 since thats what is being compared with the risk analysis score.

We need to update the properties file to change type to `Float`